dask_expr/tests/_util.py

```diff
@@ -16,3 +17,22 @@ def xfail_gpu(reason=None):
     condition = _backend_name() == "cudf"
     reason = reason or "Failure expected for cudf backend."
     return pytest.mark.xfail(condition, reason=reason)
+
+
+def assert_eq(a, b, *args, serialize_graph=True, **kwargs):
+    if serialize_graph:
+        # Check that no `Expr` instances are found in
+        # the graph generated by `Expr.dask`
+        try:
+            from distributed.protocol import serialize
+            from distributed.protocol.serialize import ToPickle
+
+            with config.set({"dask-expr-no-serialize": True}):
+                for obj in [a, b]:
+                    if hasattr(obj, "dask"):
+                        serialize(ToPickle(obj.dask))
```

Comment:
Have you tried plain pickle? While this is closer to what is happening in reality, plain pickle is likely faster for this case and tests what we're looking for

Reply:
We will currently see errors for local functions (like `lambda`s), and the run time doesn't seem to change much.

Reply:
Actually, it probably does make sense to try using `pickle.dumps` before falling back to `serialize(ToPickle(obj.dask))`. That does seem to take us below a 10% runtime difference.

Reply:
well, there is a point why this raises with lambdas :) pickle cannot deal with lambda so we'll have to fallback to cloudpickle which is more often than not slower than plain pickle because it again requires pickling by value instead of reference (i.e. it has to serialize the code of the function instead of just a pointer)

I guess this is a different topic but if dask-expr passes a plain pickle test, you'll be faster in the long run.

For instance...

```python
In [1]: import pickle, cloudpickle

In [2]: def f(x):
   ...:     return x
   ...:

In [3]: len(pickle.dumps(f))
Out[3]: 29

In [4]: len(cloudpickle.dumps(lambda x: x))
Out[4]: 455
```

when the thing needs to be pickled by value (cloudpickle) it has to write much more garbage to be able to reconstruct it. This is particularly bad if the lambda references something from the local scope since then the entire local scope is serialized.

For large graphs, this stuff can add up quickly

Reply:
Definitely good hygiene to avoid things that need cloudpickle in Dask.  This can become costly at large numbers of partitions.

Reply:
Yeah, I guess there are a few places where we were still using `lambda`s in places that we shouldn't be (I found and fixed some of those cases), but the main reason we need to fall back to `serialize` or `cloudpickle` in `assert_eq` is that we use `lambda`s in the pytests.